src/api/AppActivity.js

```diff
@@ -0,0 +1,144 @@
+// /2.0/app_activities?item_id=<FILE_ID>&item_type=file&fields=occurred_at,rendered_text,app,created_by,activity_template&marker=<NEXT_MARKER>&limit=<MY_LIMIT>
+
+/**
+ * @flow
+ * @file Helper for the box App Activity API
+ * @author Box
+ */
+import MarkerBasedAPI from './MarkerBasedAPI';
+import {
+    ERROR_CODE_DELETE_APP_ACTIVITY,
+    FIELD_ACTIVITY_TEMPLATE,
+    FIELD_APP,
+    FIELD_CREATED_BY,
+    FIELD_OCCURRED_AT,
+    FIELD_RENDERED_TEXT,
+} from '../constants';
+
+const APP_ACTIVITY_FIELDS: string = [
+    FIELD_ACTIVITY_TEMPLATE,
+    FIELD_APP,
+    FIELD_CREATED_BY,
+    FIELD_OCCURRED_AT,
+    FIELD_RENDERED_TEXT,
+].join(',');
+
+/**
+ * Map an entry from the AppActivity API to an AppActivityItem.
+ * occurred_at -> created_at
+ *
+ * @param {Object} item - A single entry in the AppActivity API entiries list
+ */
+function mapAppActivityItem(item: Object): AppActivityItem {
+    const { occurred_at, ...wOutOccurredAt } = item;
+    return {
+        created_at: occurred_at,
+        ...wOutOccurredAt,
+    };
+}
+
+class AppActivity extends MarkerBasedAPI {
+    /**
+     * Base API URL for App Activity on a file
+     *
+     * @return {string} Base API url for app activities
+     */
+    getAppActivityUrl(): string {
+        return `${this.getBaseApiUrl()}/app_activities`;
+    }
+
+    /**
+     * API URL for getting App Activity on a file
+     *
+     * @param {string} [id] - a box file id
+     * @return {string} Url for all app activity on a file
+     */
+    getUrl(id: string): string {
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
+
+        return `${this.getAppActivityUrl(id)}?item_id=${id}&item_type=file&fields=${APP_ACTIVITY_FIELDS}`;
+    }
+
+    /**
+     * API URL for deleting app activity from a file
+     *
+     * @param id - ID of an app activity item
+     * @return {string} - URL to delete app activity
+     */
+    getDeleteUrl(id: string): string {
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
+
+        return `${this.getAppActivityUrl()}/${id}`;
+    }
+
+    /**
+     * Generic success handler
+     *
+     * @param {Object} data - the response data
+     */
+    successHandler = (data: Object): void => {
```

Comment:
I think we're trying to avoid `Object` and `Function` where possible.

Reply:
👍 Moved to use AppActivityItems instead of Object.